videoanalyser/app/src/main/java/com/example/videoanalyser/MainActivity.java

```diff
@@ -0,0 +1,132 @@
+
+        filePath = findViewById(R.id.selectedFilePath);
+        filePickerBtn = findViewById(R.id.fileSelectButton);
+        analyzeBtn = findViewById(R.id.analyze);
+        recognizer = TextRecognition.getClient();
+
+        filePickerBtn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                fileIntent = new Intent(Intent.ACTION_GET_CONTENT);
+                fileIntent.setType("*/*");
+                startActivityForResult(fileIntent, 10);
+            }
+
+        });
+
+        analyzeBtn.setOnClickListener(new View.OnClickListener() {
+            @RequiresApi(api = Build.VERSION_CODES.P)
+            @Override
+            public void onClick(View view) {
+                mediaMetadataRetriever = new MediaMetadataRetriever();
+
+                try {
+                    mediaMetadataRetriever.setDataSource(getApplicationContext(), fileUri);
+                }catch (Exception e){
+                    e.printStackTrace();
+                    e.getCause();
+                }
+
+                totalFrames = Integer.valueOf(mediaMetadataRetriever.extractMetadata(METADATA_KEY_VIDEO_FRAME_COUNT));
+                List<Bitmap> frameList = mediaMetadataRetriever.getFramesAtIndex(0, Math.max(0, totalFrames-safeFrames));
+
+                for(int i=0; i<frameList.size();i++)
+                {
+                    // analysis will be done here (forking in a separate thread)
+                    imageHolder = InputImage.fromBitmap(frameList.get(i), 0);
+                    final int finalI = i;
+                    Task<Text> result =
+                            recognizer.process(imageHolder)
+                                    .addOnSuccessListener(new OnSuccessListener<Text>() {
+                                        @Override
+                                        public void onSuccess(Text visionText) {
+                                            // Task completed successfully
+                                            // ...
+                                            results.add(visionText.getText());
+                                            Log.d("Rokus logs", "Text detected at index:" + finalI + " " + visionText.getText());
+                                        }
+                                    })
+                                    .addOnFailureListener(
+                                            new OnFailureListener() {
+                                                @Override
+                                                public void onFailure(@NonNull Exception e) {
+                                                    // Task failed with an exception
+                                                    // ...
+                                                }
+                                            });
+                }
+            }
+        });
+    }
+
+    @SuppressLint("MissingSuperCall")
```

Comment:
nit: Any reason we don't also call into the superclass onActivityResult method here?
super.onActivityResult(requestCode, resultCode, data)

Reply:
It seems like I accidently deleted it and added a supress warning on compiler suggestion. Added in refactoring commit.